### PR TITLE
Support for Alias Records in Azure DNS

### PR DESCRIPTION
**Pull request by karravi-msft**
## Description
<!--
Support for new Azure DNS API version with Alias records. Swagger PR -> https://github.com/Azure/azure-rest-api-specs/pull/3777

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ x] Please add REST spec PR link to the SDK PR
- [ x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
CI is failing.
Cleanup commit history. This PR cannot be accepted with current commit history.

File: src/SDKs/Dns/Management.Dns/Microsoft.Azure.Management.Dns.csproj [hunk 1]
```diff
@@ -6,14 +6,12 @@
 	<PropertyGroup>
 		<PackageId>Microsoft.Azure.Management.Dns</PackageId>
 		<Description>Provides management capabilities for Azure Dns.</Description>
-		<Version>2.2.0-preview</Version>
+		<Version>2.3.0-preview</Version>
```
[the remark this answers is not in the document]
Comment (karravi-msft) [reply src/SDKs/Dns/Management.Dns/Microsoft.Azure.Management.Dns.csproj]: We never changed the major version previously even for new api versions as this is not a breaking api change..I will do it if needed but wondering if this is a recent policy change..

Also, previously non fluent .net sdk was always marked as preview. WIll be glad to remove the preview tag

File: src/SDKs/_metadata/dns_resource-manager.txt [hunk 2]
```diff
@@ -1,11 +1,11 @@
-2018-03-14 21:04:59 UTC
+2018-09-06 21:12:30 UTC
 
 1) azure-rest-api-specs repository information
 GitHub user: Azure
 Branch:      master
-Commit:      58409bca13e129632c26c9d3767f22f922af62dc
+Commit:      5ceb33d2e5c0692dbdbcc907e025c1dfd994f29c
```
[the remark this answers is not in the document]
Comment (karravi-msft) [reply src/SDKs/_metadata/dns_resource-manager.txt]: https://github.com/Azure/azure-rest-api-specs/pull/3777

File: src/SDKs/Dns/Management.Dns/Microsoft.Azure.Management.Dns.csproj [hunk 3]
```diff
@@ -24,5 +22,11 @@
 
 	<!-- Please do not move/edit code below this line -->
 	<Import Condition=" Exists('$([MSBuild]::GetPathOfFileAbove(AzSdk.RP.props))') " Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.RP.props'))" />
+	<ItemGroup>
```
[the remark this answers is not in the document]
Comment (karravi-msft) [reply src/SDKs/Dns/Management.Dns/Microsoft.Azure.Management.Dns.csproj]: done

File: src/SDKs/Dns/Management.Dns/Microsoft.Azure.Management.Dns.csproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (karravi-msft) [reply src/SDKs/Dns/Management.Dns/Microsoft.Azure.Management.Dns.csproj]: changed version to 3.0.0.0

**Comment by karravi-msft** [comment]
Isnt there a way to squash merge when committing the PR into github ? Anyway I tried to squash and update the same topic branch.. If this is not acceptable still , I submitted another PR - https://github.com/Azure/azure-sdk-for-net/pull/4747 with all the changes squashed into a single commit

**Comment by shahabhijeet** [comment]
Closing this PR due to the new PR https://github.com/Azure/azure-sdk-for-net/pull/4747

**Pull request closed without merging**